terminus-ui/button/src/button.component.spec.ts

```diff
@@ -1,84 +1,155 @@
+  public ngOnInit() { }
+  public ngOnDestroy() { }
+}
 
 describe(`TsButtonComponent`, function() {
-  let component: TsButtonComponent;
-
-  beforeEach(() => {
-    component = new TsButtonComponent(
-      new ChangeDetectorRefMock(),
-      new TsWindowServiceMock(),
-      new Renderer2Mock(),
-    );
-    component['changeDetectorRef'].detectChanges = jest.fn();
-    component.button = {
-      _elementRef: new ElementRef({}),
-    } as any;
-    component['renderer'].addClass = jest.fn();
-    component['renderer'].removeClass = jest.fn();
-  });
-
-
-  it(`should exist`, () => {
-    expect(component).toBeTruthy();
-  });
-
-  describe(`showProgress`, () => {
-    test(`should set and retrieve`, () => {
-      component.showProgress = true;
-      expect(component.showProgress).toEqual(true);
+
+    let component: TestHostComponent;
+    let fixture: ComponentFixture<TestHostComponent>;
+    let button: HTMLButtonElement;
+    let buttonComponent: TsButtonComponent;
+
+    beforeEach(() => {
+      fixture = createComponent(TestHostComponent, [], [TsButtonModule]);
+      component = fixture.componentInstance;
+      buttonComponent = component.buttonComponent;
+      fixture.detectChanges();
+      button = fixture.debugElement.query(By.css('.c-button')).nativeElement as HTMLButtonElement;
     });
-  });
 
-  describe(`set collapsed`, () => {
+    describe(`isDisabled`, () => {
 
-    describe(`when format === collapsable`, () => {
+      test(`button is not disabled`, () => {
+        component.disabled = false;
+        fixture.detectChanges();
+        expect(buttonComponent.isDisabled).toEqual(false);
+        expect(button.disabled).toEqual(false);
+        expect(button.getAttribute('disabled')).toEqual(null);
+      });
 
-      it(`should call collapseWithDelay if a delay is set and the value is FALSE`, () => {
-        component['collapseWithDelay'] = jest.fn();
-        component.collapseDelay = component['COLLAPSE_DEFAULT_DELAY'];
-        component.collapsed = false;
+      test(`button is disabled`, () => {
+        component.disabled = true;
+        fixture.detectChanges();
+        expect(buttonComponent.isDisabled).toEqual(true);
+        expect(button.disabled).toEqual(true);
+        expect(button.getAttribute('disabled')).toEqual('');
+      });
+    });
+
+    test(`click`, () => {
+      component.buttonComponent.clickEvent.emit = jest.fn();
+      button.click();
+      expect(buttonComponent.clickEvent.emit).toHaveBeenCalled();
+    });
 
-        expect(component['collapseWithDelay']).toHaveBeenCalled();
-        expect(component.isCollapsed).toEqual(false);
+    describe(`showProgress`, () => {
+      test(`showProgress is true`, () => {
+        component.showProgress = true;
+        fixture.detectChanges();
+        expect(buttonComponent.showProgress).toEqual(true);
+        expect(button.getAttribute('disabled')).toEqual('');
       });
 
+      test(`showProgress is false`, () => {
```

Comment:
Definitely *not* a needed change, but for future reference, feel free to combine on/off tests like these into a single `test`